File: week1/day3/threeNumSum.py
array = [12, 3, 1, 2, -6, 5, -8, 6]
target = 0

def getThreeSum(array, target):
  threeSum = []
  array.sort()
  for index, value in enumerate(array):
    a, z = index + 1, len(array) - 1
    while a < z:
      if array[a] + value + array[z] == target: threeSum.append([array[a], value, array[z]])
      if array[a] + value + array[z] > target: z -= 1
      else: a += 1
  return threeSum

print(getThreeSum(array, target))

File: week1/day3/threeNumSum.js
array = [12, 3, 1, 2, -6, 5, -8, 6]
target = 0;

const getTwoSum = ( array, target ) => {
  return array.filter( value => array.includes( (target - value) ) );
};

const getThreeSum = ( array, target ) => {
  const threeSum = []
  array.sort((a, b) => a - b)
  array.forEach( ( value, index ) => {
    let [ a, z ] = [ index + 1, array.length - 1]
    while ( a < z ) {
      switch ( array[ a ] + value + array[ z ] ) {
        case target: threeSum.push( [ array[ a ], value, array[ z ] ] );
        default: array[ a ] + value + array[ z ] < target ? a = a + 1 : z = z - 1; 
      }
    }
  } )
  return threeSum;
};

console.log( getThreeSum( array, target ) );